File: 45 Jump Game II.java
class Solution {
    public int jump(int[] nums) {
        int len = nums.length - 1;
        int curr = -1;
        int next = 0;
        int ans = 0;
        
        for (int i = 0; next < len; i++) {
            if (i > curr) {
                ans++;
                curr = next;
            };
            next = Math.max(next, nums[i] + i);
        };
        return ans;
    };
};

File: 45 Jump Game II.kt
class Solution {
    fun jump(nums: IntArray): Int {
        val mArr = IntArray(nums.size){Integer.MAX_VALUE}
        val jArr = IntArray(nums.size){Integer.MAX_VALUE}
        if(nums.size > 1) {
            mArr[0] = 0
            jArr[0] = 0
            
            for(i in 1..nums.size-1){
                for(j in 0..i-1) {
                    if( j+nums[j] >= i){
                        mArr[i] = Math.min(mArr[j] + 1, mArr[i])
                        jArr[i] = j
                    }
                }
            }
            return mArr[nums.size-1]
        } else {
            return nums.size-1
        }
    }
}
